app/src/utils/apiHooks/notebooks.tsx

```diff
@@ -52,3 +61,60 @@ export const useCreateNotebookFromTemplate = (
     mutation,
   };
 };
+
+/**
+ * Query hook to get the sync status of a set of records
+ * from a project
+ */
+export const useRecordAudit = ({
+  projectId,
+  listingId,
+  username,
+}: {
+  projectId: string;
+  listingId: string;
+  username: string;
+}) => {
+  const N = 2; // refetch time in minutes - maybe configure in env?
+  const queryKey = ['record-audit', projectId, listingId, username];
+  return useQuery({
+    queryKey,
+    queryFn: async (context: QueryFunctionContext) => {
+      const currentStatus: RecordStatus | undefined =
+        context.client.getQueryData(queryKey);
```

Comment:
Might be unnecessarily exposing tanstack details here - maybe use a useState variable instead? Not the end of the world though. Also I think you might have a different queryFn available which knows of the previous data if any.

Reply:
This seems to be the way to use the current status in a queryFn from what I can see.  That's why the context is passed in to the fn.